ansible/roles/ci/code_style_check/catkin/files/catkin_lint_xunit.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2020 Shadow Robot Company Ltd - All Rights Reserved.
+# Proprietary and Confidential. Unauthorized copying of the content in this file, via any medium is strictly prohibited.
+
+import argparse
+import os
+import shutil
+import subprocess
+import sys
+import time
+from xml.sax.saxutils import escape
+from xml.sax.saxutils import quoteattr
+
+
+def main(argv=sys.argv[1:]):
+    parser = argparse.ArgumentParser(
+        description='Check a package using catkin_lint.',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        'paths',
+        nargs='*',
+        default=[os.curdir],
+        help='The package or folder with packages to check with catkin_lint')
+    parser.add_argument(
+        '--xunit-file',
+        help='The path where a xunit compliant XML file should be created')
+    parser.add_argument(
+        '--lintignore',
+        help='Name of the lintignore file for catkin (normally .catkin_lint_ignore)')
+    args = parser.parse_args(argv)
+    packages = args.paths
+    if not packages:
+        print('No packages found', file=sys.stderr)
+        return 1
+    packages = [os.path.abspath(package) for package in packages]
+    error_report=run_catkin_lint(packages,args.lintignore)
+    if args.xunit_file:
+        start_time = time.time()
+    generate_xunit_file(error_report,args.xunit_file,start_time)
+
+
+def run_catkin_lint(packages, lintignore):
+    report = []
+    catkinlint_bin = shutil.which('catkin_lint')
+    if not catkinlint_bin:
```

Comment:
not sure about this, how is it dealt with if this is returned, the report will be this string? maybe it should be in the main and return 1 instead?

Reply:
Fixed in the last commit